File: frontend/src/lib/api/customer-portal.ts
/**
 * 고객 포털 API (인증 없음)
 */

import axios from "axios";

const API_URL = process.env.NEXT_PUBLIC_API_URL || "http://localhost:8000/api/v1";

// 인증 없이 사용하는 클라이언트
const publicClient = axios.create({
  baseURL: API_URL,
  headers: {
    "Content-Type": "application/json",
  },
});

// ===== Types =====

export interface CustomerViewPhoto {
  url: string;
  filename: string;
}

export interface CustomerViewQuoteItem {
  item_name: string;
  description: string | null;
  quantity: number;
  unit: string;
  unit_price: number;
  amount: number;
}

export interface CustomerViewQuote {
  quote_number: string;
  quote_date: string;
  items: CustomerViewQuoteItem[];
  total_amount: number;
  estimate_note: string | null;
  pdf_download_url: string;
}

export interface ProgressStep {
  step: string;
  status: "completed" | "current" | "pending" | "cancelled";
  date: string | null;
}

export interface CustomerViewResponse {
  assignment_id: number;
  assignment_status: string;
  status_label: string;
  assigned_services: string[];
  scheduled_date: string | null;
  scheduled_time: string | null;
  application_number: string;
  customer_name_masked: string;
  address_partial: string;
  selected_services: string[];
  description: string | null;
  partner_company: string | null;
  partner_phone_masked: string | null;
  quote: CustomerViewQuote | null;
  work_photos_before: CustomerViewPhoto[];
  work_photos_after: CustomerViewPhoto[];
  work_photos_uploaded_at: string | null;
  progress_steps: ProgressStep[];
  created_at: string;
  token_expires_at: string;
  contact_info: string;
}

export interface CustomerViewTokenInfo {
  is_valid: boolean;
  expires_at: string | null;
  assignment_id: number | null;
  message: string | null;
}

// ===== API Functions =====

/**
 * 고객 열람 데이터 조회
 */
export async function getCustomerView(token: string): Promise<CustomerViewResponse> {
  const response = await publicClient.get<CustomerViewResponse>(
    `/customer-portal/view/${token}`
  );
  return response.data;
}

/**
 * 토큰 유효성 확인
 */
export async function getTokenInfo(token: string): Promise<CustomerViewTokenInfo> {
  const response = await publicClient.get<CustomerViewTokenInfo>(
    `/customer-portal/token-info/${token}`
  );
  return response.data;
}

/**
 * 견적서 PDF 다운로드 URL 생성
 */
export function getQuotePdfUrl(token: string): string {
  return `${API_URL}/customer-portal/quote-pdf/${token}`;
}

File: backend/app/schemas/customer_portal.py
"""
Customer Portal schemas
고객 열람 포털 - 시공 정보 및 사진 열람용 스키마
"""

from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class CustomerViewPhoto(BaseModel):
    """사진 정보 (토큰화된 URL)"""
    url: str
    filename: str


class CustomerViewQuoteItem(BaseModel):
    """견적 항목"""
    item_name: str
    description: Optional[str] = None
    quantity: float
    unit: str
    unit_price: int
    amount: int


class CustomerViewQuote(BaseModel):
    """견적 정보"""
    quote_number: str
    quote_date: str
    items: list[CustomerViewQuoteItem]
    total_amount: int
    estimate_note: Optional[str] = None
    pdf_download_url: str


class CustomerViewResponse(BaseModel):
    """고객 열람용 시공 정보 응답 (개인정보 마스킹)"""

    # 배정 정보
    assignment_id: int
    assignment_status: str
    status_label: str  # 상태 한글 표시
    assigned_services: list[str]
    scheduled_date: Optional[str] = None
    scheduled_time: Optional[str] = None

    # 신청 정보 (마스킹 적용)
    application_number: str
    customer_name_masked: str  # 홍** 형식
    address_partial: str  # 동/리까지만 표시
    selected_services: list[str]  # 전체 서비스 목록
    description: Optional[str] = None

    # 협력사 정보 (연락처 마스킹)
    partner_company: Optional[str] = None
    partner_phone_masked: Optional[str] = None  # 010-****-5678 형식

    # 견적 정보
    quote: Optional[CustomerViewQuote] = None

    # 시공 사진 (토큰화된 URL)
    work_photos_before: list[CustomerViewPhoto] = []
    work_photos_after: list[CustomerViewPhoto] = []
    work_photos_uploaded_at: Optional[str] = None

    # 진행 현황
    progress_steps: list[dict]  # [{"step": "접수", "status": "completed", "date": "..."}]

    # 메타 정보
    created_at: str
    token_expires_at: str
    contact_info: str = "문의: 031-XXX-XXXX"


class CustomerViewTokenInfo(BaseModel):
    """토큰 정보 응답"""
    is_valid: bool
    expires_at: Optional[str] = None
    assignment_id: Optional[int] = None
    message: Optional[str] = None
